### PR TITLE
feat(logging) - Errors and Logging

### PR DESCRIPTION
**Pull request by akintolga**
- [x] [SD-1297] Re-factoring the API (HTTP) Errors: 
- [x] [SD-1274] Backend: Create a list of errors (with error codes) to be raised in ingest
- [x] [SD-1275] Backend: Create log messages to changes to channel configurations (Mayur has a change for this and I added more logging for errors)

### CODE REVIEW

File: superdesk/errors.py [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+    @classmethod
+    def anpaError(cls, exception):
+        return ProviderError(2005, exception)
+
+
+class ParserError(SuperdeskIngestError):
+    _codes = {
+        1001: 'Message could not be parsed',
+        1002: 'Ingest file could not be parsed',
+        1003: 'ANPA file could not be parsed',
```
Comment (akintolga) [inline superdesk/errors.py]: There are two reasons why I implemented a global errors.py module is that:
* One requirement to get the available errors to configuration UI at some point to allow administrators/users to choose which errors they want to get notified for.
* The other reason was to see the whole hierarchy of errors and the available codes reduce redundancy of types and messages.

I can see that if another customer has nothing to do with -say- ANPA parsing why would we have error code? I'd say even if the ANPA is not there we will have the error codes 1001, 1002, 1004 ... will be there so the users could ask why 1003 is missing.

**Comment by amagdas** [comment]
Is there anything else that needs to be done for this PR?

**Comment by akintolga** [comment]
HI Adrian, that's all for the time being. If you don't see any major problems then there's nothing else.
We may need to look after the ingest the first few days after the merge to see anything is broken. I'll do the same on this side on our servers as well.

**Comment by amagdas** [comment]
Looks good to me :+1:

**Comment by petrjasek** [comment]
:+1:

**Pull request closed without merging**